### PR TITLE
Add browser API wrappers

### PR DESCRIPTION
**Pull request by lesterfan**
### Description

This adds Python wrappers for `setTimeout`, `clearTimeout`, `setInterval`, `clearInterval`, `addEventListener` and `removeEventListener` described in https://github.com/pyodide/pyodide/issues/2430.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by hoodmane** [comment]
@lesterfan I fixed the tests in chrome and firefox, they should run locally for you. Node still can't find `clearInterval`, etc.

**Comment by lesterfan** [comment]
@hoodmane I just made the change to implement these wrappers in Python using `elt.js_id` for `add_event_listener`/`remove_event_listener` and fixed the resource leaks in my tests. As discussed, I'll leave fixing the imports in node to you. Let me know if there's something else I should do; I'm happy to help however I can to get this merged!

**Comment by hoodmane** [comment]
@lesterfan I fixed the node tests. Turns out in node, `setTimeout` returns an object with some information rather than an int. So there needs to be some special handing for this.

**Comment by lesterfan** [comment]
Ah, I see. I just pushed a change which (I think) will handle this for node. Locally, the node tests time out at `test_clear_timeout_destroyable_noop` and I'm not sure why, especially since the first two tests using `set_timeout` and `clear_timeout` pass.

File: src/py/pyodide/_browser_apis.py [hunk 1]
```diff
@@ -53,7 +54,8 @@ def wrapper():
 DUMMY_DESTROYABLE = Destroyable()
 
 
-def clear_timeout(id: int):
+def clear_timeout(timeout_retval: int | JsProxy):
+    id = timeout_retval if isinstance(timeout_retval, int) else timeout_retval.js_id
     clearTimeout(id)
```
Comment (hoodmane) [inline src/py/pyodide/_browser_apis.py:59]: This needs to be `clearTimeout(timeout_retval)`.

File: src/py/pyodide/_browser_apis.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+def set_interval(callback: Callable[[], None], interval: int) -> int:
+    id = -1
+
+    def wrapper():
+        nonlocal id
+        callback()
+        TIMEOUTS.pop(id, None)
+
+    callable = create_once_callable(wrapper)
```
Comment (hoodmane) [inline src/py/pyodide/_browser_apis.py:71]: I'm pretty sure you will need the same change for `setInterval` and `clearInterval`.

Comment (lesterfan) [reply src/py/pyodide/_browser_apis.py:71]: Ah, good point. I also realized I forgot to add tests for those two functions as well.

**Comment by lesterfan** [comment]
@hoodmane When adding tests for `set_interval` I ran into this error when testing: ```
pyodide.asm.js:10 Uncaught Error: OnceProxy can only be called once
    at wrapper (pyodide.asm.js:10:246569)
```. Any ideas on how to go about fixing this?

**Comment by hoodmane** [comment]
> Any ideas on how to go about fixing this?

We need to use `create_proxy` instead of `create_once_callable` in `set_interval`.

**Comment by hoodmane** [comment]
Actually, `set_interval` and `clear_interval` should be like `add_event_listener` and `remove_event_listener` not like `set_timeout` and `clear_timeout`.

**Comment by lesterfan** [comment]
Sounds good; I just made the change to `set_interval` and `clear_interval`.

**Comment by hoodmane** [comment]
Okay, code looks good and CI passes. Could you add a changelog entry and docstrings for the new functions? Then we will merge this.

**Comment by hoodmane** [comment]
Thanks @lesterfan!

**Pull request merged**